### DISCUSSION TITLE
TypeScript: `CommonProps['setValue']` doesn't match `Select.tsx`'s `setValue`

### DISCUSSION THREAD

**Discussion by vjee** [JedWatson/react-select#4877] [Q&A] [answered]
After upgrading to 5.1, I have noticed that the type definition of [`CommonProps['setValue']`](https://github.com/JedWatson/react-select/blob/master/packages/react-select/src/types.ts#L89-L93) does not match the implementation in [Select.tsx](https://github.com/JedWatson/react-select/blob/master/packages/react-select/src/Select.tsx#L894-L898).

```ts
// CommonProps['setValue']
setValue: (
  newValue: OnChangeValue<Option, IsMulti>,
  action: SetValueAction,
  option: Option
) => void;

// Implementation inside Select.tsx
setValue = (
  newValue: OnChangeValue<Option, IsMulti>,
  action: SetValueAction,
  option?: Option
) => {
```

Notice how the third argument `options` is optional in `Select.tsx`.

Is this intended?
At the moment I have to resort to passing `undefined as any`:

```tsx
import Select, { MenuListProps } from 'react-select';

<Select components={{ MenuList }} {...otherPropsHere} />

function MenuList(props: MenuListProps) {
  const { setValue } = props;

  // Somewhere down the line:
  setValue(newValue, 'select-option', undefined as any);
}
```

If this is not intended behaviour, I'd be happy to try and fix this in a PR.
Or am I using it incorrectly?

**Comment by Methuselah96** [comment, marked as answer]
Thanks, good catch! It should be optional. The intention is that for multi-select the option will be passed, but it won't be for single-select. If you could make a PR that would be great.

**Reply by vjee** [reply]
Alright, this should do it: https://github.com/JedWatson/react-select/pull/4908
